core/dqn_agent.py

```
@@ -0,0 +1,221 @@
+import os
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import random
+from collections import deque
```

Ruff (0.12.2)

`collections.deque` imported but unused

Remove unused import: `collections.deque`

(F401)